src/fmpz_mod_mpoly/test/t-compose_fmpz_mod_mpoly.c

```diff
@@ -0,0 +1,381 @@
+        fmpz_mod_mpoly_init(A, ctxAC);
+        fmpz_mod_mpoly_init(A1, ctxAC);
+        fmpz_mod_mpoly_init(A2, ctxAC);
+        for (i = 0; i < 3; i++)
+        {
+            Cp[i] = C + i;
+            fmpz_mod_mpoly_init(C + i, ctxAC);
+        }
+
+        fmpz_mod_mpoly_set_str_pretty(B,
+                "1 + x1*x2^2 + x2^9999999999999999999999999*x3^9", NULL, ctxB);
+
+        fmpz_mod_mpoly_set_str_pretty(C + 0, "x1 + x2", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 1, "x1 - x2", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 2, "1", NULL, ctxAC);
+        if (fmpz_mod_mpoly_compose_fmpz_mod_mpoly(A, B, Cp, ctxB, ctxAC) ||
+            fmpz_mod_mpoly_compose_fmpz_mod_mpoly_horner(A1, B, Cp, ctxB, ctxAC) ||
+            fmpz_mod_mpoly_compose_fmpz_mod_mpoly_geobucket(A2, B, Cp, ctxB, ctxAC))
+        {
+            printf("FAIL\n");
+            flint_printf("Check non-example 1\n", i);
+            fflush(stdout);
+            flint_abort();
+        }
+
+        fmpz_mod_mpoly_set_str_pretty(C + 0, "x1", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 1, "2*x2", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 2, "1", NULL, ctxAC);
+        if (!fmpz_mod_mpoly_compose_fmpz_mod_mpoly(A, B, Cp, ctxB, ctxAC) ||
+            !fmpz_mod_mpoly_compose_fmpz_mod_mpoly_horner(A1, B, Cp, ctxB, ctxAC) ||
+            !fmpz_mod_mpoly_compose_fmpz_mod_mpoly_geobucket(A2, B, Cp, ctxB, ctxAC))
+        {
+            printf("FAIL\n");
+            flint_printf("Check non-example 2\n", i);
+            fflush(stdout);
+            flint_abort();
+        }
+
+        fmpz_mod_mpoly_set_str_pretty(C + 0, "2*x1", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 1, "x2", NULL, ctxAC);
+        fmpz_mod_mpoly_set_str_pretty(C + 2, "1", NULL, ctxAC);
+        if (!fmpz_mod_mpoly_compose_fmpz_mod_mpoly(A, B, Cp, ctxB, ctxAC) ||
+            !fmpz_mod_mpoly_compose_fmpz_mod_mpoly_horner(A1, B, Cp, ctxB, ctxAC) ||
+            !fmpz_mod_mpoly_compose_fmpz_mod_mpoly_geobucket(A2, B, Cp, ctxB, ctxAC))
+        {
+            printf("FAIL\n");
+            flint_printf("Check example 3\n", i);
+            fflush(stdout);
+            flint_abort();
+        }
+
+        fmpz_mod_mpoly_clear(B, ctxB);
+        fmpz_mod_mpoly_clear(A, ctxAC);
+        fmpz_mod_mpoly_clear(A1, ctxAC);
+        fmpz_mod_mpoly_clear(A2, ctxAC);
+        for (i = 0; i < 3; i++)
+            fmpz_mod_mpoly_clear(C + i, ctxAC);
+
+        fmpz_mod_mpoly_ctx_clear(ctxB);
+        fmpz_mod_mpoly_ctx_clear(ctxAC);
+    }
+
+    /* check composition with generators */
+    for (i = 0; i < 20*flint_test_multiplier(); i++)
+    {
+        fmpz_t p;
+        fmpz_mod_mpoly_ctx_t ctxB, ctxAC;
+        fmpz_mod_mpoly_t B, A, A1, A2, A3;
+        fmpz_mod_mpoly_struct ** C;
+        slong * c;
+        slong nvarsB, nvarsAC;
+        slong len;
+        flint_bitcnt_t exp_bits;
+
+        fmpz_init_set_ui(p, 100003);
```

Comment:
Why this specific modulus?

Reply:
No particularly reason. I believe it came from the `t-resultant_discriminant.c` test. I've changed them to be consistent with the other test. They all now use 13.

Reply:
Hmm, I believe it would be better to randomize it. You could do something along the lines of
```c
fmpz_init(p);
fmpz_randprime(p, state, n_randint(state, 15), 0);
```

Reply:
Done, had to use +2 to prevent trying to generate a prime less than 2